File: src/Python/686.RepeatedStringMatch.py

class Solution:
    def repeatedStringMatch(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: int
        """
        count = len(B) // len(A)
        m = A * count
        if B in m:
            return count
        m += A
        if B in m:
            return count + 1
        m += A
        if B in m:
            return count + 2
        return -1
                
if __name__ == '__main__':
    solution = Solution()
    print(solution.repeatedStringMatch("abcd", "cdabcdab"))
else:
    pass

File: src/Java/601-700/686.RepeatedStringMatch.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int repeatedStringMatch(String A, String B) {
        int count = B.length() / A.length();
        StringBuilder strBuilder = new StringBuilder();
        for (int i = 0; i < count; i++) {
            strBuilder.append(A);
        }
        String match = strBuilder.toString();
        if (match.contains(B) == true) 
            return count;
        count += 1;
        match += A;
        if (match.contains(B) == true) 
            return count;
        count += 1;
        match += A;
        if (match.contains(B) == true) 
            return count;
        return -1;
    }
}
